src/klein/resource.py

```diff
@@ -56,6 +60,5 @@ def __repr__(self):
         return "<special bound method/module klein.resource>"
 
 
-preserve = modules[__name__]
-modules[__name__] = _SpecialModuleObject()  # type: ignore
-modules[__name__].__preserve__ = preserve  # type: ignore
```

Comment:
Rather than assigning to an attribute after the fact, let's pass in the value in the constructor.